Shopipy.Shared/Services/IProductService.cs

```diff
@@ -0,0 +1,12 @@
+﻿using Shopipy.Persistence.Models;
+
+namespace Shopipy.Shared.Services
+{
+    public interface IProductService
+    {
+        Task<Product> CreateProductAsync(Product product, int businessId);
+        Task<Product?> GetProductByIdAsync(int productId, int businessId);
+        Task<Product?> UpdateProductAsync(int productId, Product product, int businessId);
```

Comment:
Why are these two nullable?

Reply:
`GetProductByIdAsync` makes sense to return nullable as product may not exist.`UpdateProductAsync` doesn't make sense, also why do you need to specify additional ids in update product? Shouldn't they already be in the `Product` you pass?

Reply:
@piatrashkakanstantinass your comment makes sense, yes. But by the contract, in the request body, we have the product without ids. Same thing with variations.

Reply:
@Domarkaite don't quite follow, existing product you pass to update should already have id and business id
```cs
            var existingProduct = await _productService.GetProductByIdAsync(productId, businessId);

            if (existingProduct == null) return NotFound();

            _mapper.Map(dto, existingProduct);
            existingProduct.UpdatedAt = DateTime.UtcNow;

            var updatedProduct = await _productService.UpdateProductAsync(productId, existingProduct, businessId);
```

Reply:
look at the request body in swagger